File: 준영/프로그래머스/LV2/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort(reverse=True)
    start = 0
    end = len(people) - 1

    while start < end:
        sum_boat = people[start] + people[end]
        if sum_boat > limit:
            start += 1
        else:
            start += 1
            end -= 1
        answer += 1

    if start == end: answer += 1
    
    return answer

File: 준영/프로그래머스/LV2/구명보트.js
function solution(people, limit){
	var answer = 0
  people.sort((a,b) => b-a)
  let l = 0
  let r = people.length-1
  
  while(l<r){
    var sum = people[l] + people[r]
    if(sum>limit){
      l += 1
    } else {
      l += 1
      r -= 1
    }
    answer += 1
  }
  if (l == r) answer += 1
  return answer
}